spec/virtual_includes_spec.rb

```diff
@@ -228,6 +240,78 @@
     end
   end
 
+  context ".merge_includes" do
+    it "merges when first is blank" do
+      first  = {}
+      second = {:key => {}}
+      result = {:key => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges when second is blank" do
+      first  = {:key => {}}
+      second = {}
+      result = {:key => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "keeps other keys" do
+      first  = {:other1 => {}}
+      second = {:other2 => {}}
+      result = {:other1 => {}, :other2 => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges blank, blank" do
+      first  = {:key => {}}
+      second = {:key => {}}
+      result = {:key => {}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges blank, hash" do
+      first  = {:key => {}}
+      second = {:key => {:more2 => {}}}
+      result = {:key => {:more2 => {}}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges hash, blank" do
+      first  = {:key => {:more1 => {}}}
+      second = {:key => {}}
+      result = {:key => {:more1 => {}}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges hash, hash - 1 level only" do
+      first  = {:key => {:more1 => {}}}
+      second = {:key => {:more2 => {}}}
+      result = {:key => {:more1 => {}, :more2 => {}}}
+      expect(Author.merge_includes(first, second)).to eq(result)
+    end
+
+    it "merges hash(hash), hash(symbol)" do
+      first  = {:key => {:more1 => {}}}
+      second = {:key => :more2}
+      result = {:key => {:more1 => {}, :more2 => {}}}
```

Comment:
Interesting, so the value of the `:more2` key depends on the value of the`:more1` key?

Reply:
We converted `second` from `{:key => :more2}` to `{:key => {:more2 => {}}}`

Reply:
In our code:

- we mostly pass the second format for `includes()` (the pure hash version).
- we mostly pass the first format for `:uses` (symbols or hashes point to symbols)